### ISSUE TITLE
typed-argument-parser is uninstallable with "tap.py", which uses the same package name

### ISSUE THREAD

**Issue by juliangilbey** [swansonk14/typed-argument-parser#67]
This looks like a really excellent and much-needed package, thank you!

Then I thought: I wonder if it's in Debian? But to my dismay, I discovered that Debian already has [`tap.py`](https://pypi.org/project/tap.py/) in the archive (in the package `python3-tap`. Since both `tap.py` and `typed-argument-parser` provide the Python package `tap`, they cannot be simultaneously installed in any environment, and `tap.py` seems generic enough that one might wish to use both simultaneously.

Unfortunately, I don't have an obvious suggestion for how to resolve this, but I wanted to bring it to your attention.

**Comment by juliangilbey** [comment]
Actually, there is a possibility: rename the Python module to `typed_argument_parser` instead of  `tap`.  Since it's only used in the import section, the longer name will make almost no difference.  Obviously you'd have to bump the version number to 2.0 to do this. (And you can't call it `typed_argparse` as that name is also taken!)

**Comment by swansonk14** [comment]
Hi @juliangilbey,

Thank you for bringing up this issue! We realize this is potentially problematic, but in order to preserve backward compatibility, we think it would be best to stick with the current naming. Since `tap.py` is a relatively small package with a very different purpose, we hope that this will not be an issue for most users.

Best,
Jesse and Kyle

**Comment by juliangilbey** [comment]
Yes, indeed, but it makes including it in Debian and other Linux distributions hard because of the package conflict.  I also just stumbled upon PEP 423: https://www.python.org/dev/peps/pep-0423/#use-a-single-name

Anyway, I would imagine an upgrading path something like the following:
* V 1.8.0 provides two Python modules: `tap` and `typed_argument_parser`.  All of the functionality is moved to the new module, and `tap` just re-exports everything from `typed_argument_parser` with a `DeprecationWarning`.
* Some time later (maybe a year or so), V 2.0.0 drops the `tap` module and only provides `typed_argument_parser`.

Or perhaps these version numbers should be V 2.0.0 and V 2.1.0 instead, so that the `DeprecationWarning` is only issued on a new major version.

Then packages which depend on `typed_argument_parser >=1.0.0,<2.0.0` will be fine.

Best wishes,

Julian

**Comment by swansonk14** [comment]
Hi @juliangilbey,

We agree that it would be really nice to have a single name that does not conflict with `tap.py`. However, we think that keeping backwards compatibility and the simplicity of writing `from tap import Tap` is enough of a reason to keep the current name. Also, other majors projects have multiple names (e.g., `scikit-learn` and `sklearn`). We're sorry if this causes you any inconvenience!

Best,
Jesse and Kyle
